src/undate/undate.py

```
@@ -506,23 +509,45 @@ def duration(self) -> Timedelta | UnDelta:
-                return latest - self.earliest + ONE_DAY
+                    year_days = (year_end - year_start).days + 1
+                    possible_max_days.add(year_days)
 
-        # TODO: handle year precision + unknown/partially known year
-        # (will be handled in separate branch)
+        # if there is more than one possible value for number of days
+        # due to range including lear year / non-leap year, return an uncertain delta
+        if possible_max_days and len(possible_max_days) > 1:
+            return UnDelta(*possible_max_days)
+        else:
+            return Timedelta(possible_max_days.pop())
```

Pylint (3.3.7)

[refactor] Unnecessary "else" after "return", remove the "else" and de-indent the code inside it

(R1705)